MQ2FeedMe.cpp

```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
+	{
+		ReadList(&Hunger, "FOOD");
+		ReadList(&Thirst, "DRINK");
 	}
-	else if (!_stricmp(Arg, "announceConsume")) {
-		if (bAnnConsume) {
+	else if (ci_equals(Arg, "announceConsume"))
+	{
+		if (bAnnConsume)
+		{
 			bAnnConsume = false;
 			WriteChatf("\ay%s\aw::Consumption Notification Off", PLUGIN_NAME);
 		}
-		else {
+		else
+		{
 			bAnnConsume = true;
 			WriteChatf("\ay%s\aw::Consumption Notification On", PLUGIN_NAME);
 		}
 	}
+	else if (!_stricmp("ui", Arg) || !_stricmp("gui", Arg))
+	{
+		DoCommand("/mqsettings plugins/feedme");
+	}
 }
 
-void AutoDrinkCmd(PSPAWNINFO pLPlayer, char* szLine)
+void AutoFeedCmd(PlayerClient* pLPlayer, char* szLine)
 {
-	if (!strlen(szLine)) {
-		if (GoodToFeed()) Consume(ItemClass_Drink, Thirst);
+	if (!strlen(szLine))
+	{
+		if (GoodToConsume())
+		{
+			Consume(ItemClass_Food, Hunger);
+		}
 		return;
 	}
-	CHAR Arg[MAX_STRING] = { 0 };
+
+	char Arg[MAX_STRING] = { 0 };
 	GetArg(Arg, szLine, 1);
-	if (!_stricmp(Arg, "list")) {
-		WriteChatf("\ay%s\aw:: Listing Drink:", PLUGIN_NAME);
-		ListTypes(Thirst);
-		if (bAnnLevels) {
-			sprintf_s(Buffer, "%d", DrnkAt);
-			WriteChatf("\ay%s\aw:: AutoDrink(\ag%s\ax).", PLUGIN_NAME, (DrnkAt) ? Buffer : "\aroff");
-			WriteChatf("\ay%s\aw:: Current Thirst(\ag%d\ax)", PLUGIN_NAME, GetPcProfile()->thirstlevel);
-		}
-	} else if (!_stricmp(Arg, "warn")) {
-		if (bDrinkWarn) {
-			bDrinkWarn = 0;
-			WriteChatf("\ay%s\aw:: Drink Warning Off", PLUGIN_NAME);
-		} else {
-			bDrinkWarn = 1;
-			WriteChatf("\ay%s\aw:: Drink Warning On", PLUGIN_NAME);
+
+	if (ci_equals(Arg, "list")) {
+		WriteChatf("\ay%s\aw:: Listing Food:", PLUGIN_NAME);
+		ListTypes(Hunger);
+
+		if (bAnnLevels)
+		{
+			sprintf_s(Buffer, "%d", iFeedAt);
+			WriteChatf("\ay%s\aw:: AutoFeed (\ag%s\ax).", PLUGIN_NAME, (iFeedAt) ? Buffer : "\aroff");
+			WriteChatf("\ay%s\aw:: Current Hunger (\ag%d\ax)", PLUGIN_NAME, GetPcProfile()->hungerlevel);
 		}
 	}
-	else if (!_stricmp(Arg, "reload")) {
-		ReadList(&Hunger, "FOOD");
-		ReadList(&Thirst, "DRINK");
+	else if (ci_equals(Arg, "warn"))
+	{
+		if (bFoodWarn)
+		{
+			bFoodWarn = 0;
+			WriteChatf("\ay%s\aw:: Food Warning Off", PLUGIN_NAME);
+		}
+		else
+		{
+			bFoodWarn = 1;
+			WriteChatf("\ay%s\aw:: Food Warning On", PLUGIN_NAME);
+		}
+	}
+	else if (ci_equals(Arg, "add"))
+	{
+		HandleFoodDrinkItem();
+	}
+	else if (IsNumber(Arg))
+	{
+		iFeedAt = GetIntFromString(Arg, 5000);
+
+		if (iFeedAt < 0)
+		{
+			iFeedAt = 0;
+		}
+		else if (iFeedAt > 5000)
+		{
+			iFeedAt = 5000;
+		}
+
+		sprintf_s(Buffer, "%d", iFeedAt);
+		WritePrivateProfileString(GetCharInfo()->Name, "AutoFeed", Buffer, INIFileName);
+		WriteChatf("\ay%s\aw:: AutoFeed (\ag%s\ax).", PLUGIN_NAME, (iFeedAt) ? Buffer : "\aroff");
+
+		if (bAnnLevels)
+		{
+			WriteChatf("\ay%s\aw:: Current Thirst (\ag%d\ax) Hunger (\ag%d\ax)", PLUGIN_NAME, GetPcProfile()->thirstlevel, GetPcProfile()->hungerlevel);
+		}
+	}
+
+	GenericCommand(szLine);
+}
+
+void AddDrinkItem()
```

Comment:
Why do we handle food but add drink? (I'll be honest, I didn't look too closely at either one nor did I compare the old one to the new)